modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/CoreInstance.java

```diff
@@ -38,97 +38,145 @@
     private CoreInstance() {
     }
 
+    /**
+     * Gets a {@link CoreSession} for the currently logged-in user.
+     *
+     * @param repositoryName the repository name, or {@code null} for the default repository
+     * @return the session
+     * @since 11.1
+     */
+    public static CoreSession getCoreSession(String repositoryName) {
+        return getCoreSession(repositoryName, getPrincipal(null));
+    }
+
+    /**
+     * MUST ONLY BE USED IN UNIT TESTS to get a {@link CoreSession} for the given user.
```

Comment:
Yes this will have to be cleaned up in the future. For now I'm keeping it also because it's used by the deprecated `openCoreSession` signature that delegates to it.
Also there's still one non-unit-test use in the CMIS code that I need to review.
Finally I'm not sure where to move it...